src/datasources/tag/TagDataSource.ts

```diff
@@ -25,27 +25,53 @@ export class TagDataSource extends DataSourceBase<TagQuery> {
   tagUrl = this.instanceSettings.url + '/nitag/v2';
   tagHistoryUrl = this.instanceSettings.url + '/nitaghistorian/v2/tags';
 
-  defaultQuery = {
+  defaultQuery: Omit<TagQuery, 'refId'> = {
     type: TagQueryType.Current,
     path: '',
     workspace: '',
     properties: false,
   };
 
   async runQuery(query: TagQuery, { range, maxDataPoints, scopedVars }: DataQueryRequest): Promise<DataFrameDTO> {
-    const { tag, current } = await this.getLastUpdatedTag(
+    const tagsLastUpdates: TagWithValue[] = await this.getLastUpdatedTag(
       this.templateSrv.replace(query.path, scopedVars),
       this.templateSrv.replace(query.workspace, scopedVars)
     );
 
+    const tag = tagsLastUpdates[0].tag
     const name = tag.properties?.displayName ?? tag.path;
     const result: DataFrameDTO = { refId: query.refId, fields: [] };
 
     if (query.type === TagQueryType.Current) {
+      const allPossibleProps = this.getAllProperties(tagsLastUpdates);
       result.fields = [
-        { name, values: [this.convertTagValue(tag.type ?? tag.datatype, current?.value.value)] },
-        { name: 'updated', values: [current?.timestamp], type: FieldType.time, config: { unit: 'dateTimeFromNow' } },
-      ];
```

Comment:
Having separate fields for the name and value of a tag is technically a breaking change. The data frame structure changes as so:

| my.tag | updated      |
|--------|--------------|
| 3.14   | a minute ago |

-> 

| name   | currentValue | updated      |
|--------|--------------|--------------|
| my.tag | 3.14         | a minute ago |

While simple use cases should be unaffected, it's possible that this could break existing dashboards. We need to decide if we're okay with the breaking change or make the new behavior opt-in.

@joshuaprewitt @prestwick

Reply:
I would recommend 16 or 32 so that a user could all of the CPUs for a single system or all of the analog inputs for a DAQ board.

Reply:
If it is the right thing to do in the long run I’m ok with the breaking change.

Reply:
If we can get the change in early for the May release I would be happy to throughly test it out just to be safe.